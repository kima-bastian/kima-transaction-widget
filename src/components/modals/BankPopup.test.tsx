import { fireEvent, render, screen } from '@testing-library/react'
import React from 'react'
import { BankPopup } from '.'
import { KimaProvider } from '../..'

describe('Bank modal test', function () {
  it('click submit button', function () {
    const { container } = render(
      <KimaProvider>
        <BankPopup />
      </KimaProvider>
    )
    expect(screen.getByText('Submit')).toBeInTheDocument()

    const submitButton = container.querySelector('.bank-simulation button')
    fireEvent.click(submitButton as Element)

    expect(screen.getByText('Transferring funds ...')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('clicking overlay should close modal', function () {
    const { container } = render(
      <KimaProvider>
        <BankPopup />
      </KimaProvider>
    )

    fireEvent.click(container.querySelector('.modal-overlay') as Element)
    expect(container).toMatchSnapshot()
  })

  it('click close button', function () {
    const { container } = render(
      <KimaProvider>
        <BankPopup />
      </KimaProvider>
    )

    fireEvent.click(container.querySelector('.topbar .icon-button') as Element)
    expect(container).toMatchSnapshot()
  })
})
