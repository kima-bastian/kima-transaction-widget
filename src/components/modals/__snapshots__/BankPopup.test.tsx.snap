// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bank modal test click close button 1`] = `
<div>
  <div
    class="kima-modal bank-popup undefined "
  >
    <div
      class="modal-overlay"
    />
    <div
      class="modal-content-container"
    >
      <div
        class="kima-card-header"
      >
        <div
          class="topbar"
        >
          <div
            class="title"
          />
          <div
            class="control-buttons"
          >
            <button
              class="icon-button"
            >
              <svg
                height="32"
                viewBox="0 0 11 11"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.656 1.688L5.944 5.4L9.656 9.112L8.712 10.056L5 6.344L1.288 10.056L0.344 9.112L4.056 5.4L0.344 1.688L1.288 0.743999L5 4.456L8.712 0.743999L9.656 1.688Z"
                  fill="white"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="modal-content"
      >
        <div
          class="bank-simulation"
        >
          <h1>
            Welcome to the First National Crypto Bank
          </h1>
          <p>
            You’re about to send funds to the Kima Finance LTD. bank account
          </p>
          <div
            class="content-item"
          >
            <span>
              User account:
            </span>
            <p>
              2345234525245
            </p>
          </div>
          <div
            class="content-item"
          >
            <span>
              Sum:
            </span>
            <p>
              $
              0
            </p>
          </div>
          <div
            class="content-item"
          >
            <span>
              Memo:
            </span>
            <p>
              795c52aa3017d3415c8da49b905e035e
            </p>
          </div>
          <button
            class="primary-button undefined"
          >
            Submit
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Bank modal test click submit button 1`] = `
<div>
  <div
    class="kima-modal bank-popup undefined "
  >
    <div
      class="modal-overlay"
    />
    <div
      class="modal-content-container"
    >
      <div
        class="kima-card-header"
      >
        <div
          class="topbar"
        >
          <div
            class="title"
          />
          <div
            class="control-buttons"
          >
            <button
              class="icon-button"
            >
              <svg
                height="32"
                viewBox="0 0 11 11"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.656 1.688L5.944 5.4L9.656 9.112L8.712 10.056L5 6.344L1.288 10.056L0.344 9.112L4.056 5.4L0.344 1.688L1.288 0.743999L5 4.456L8.712 0.743999L9.656 1.688Z"
                  fill="white"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="modal-content"
      >
        <div
          class="bank-simulation"
        >
          <h1>
            Welcome to the First National Crypto Bank
          </h1>
          <p>
            You’re about to send funds to the Kima Finance LTD. bank account
          </p>
          <div
            class="content-item"
          >
            <span>
              User account:
            </span>
            <p>
              2345234525245
            </p>
          </div>
          <div
            class="content-item"
          >
            <span>
              Sum:
            </span>
            <p>
              $
              0
            </p>
          </div>
          <div
            class="content-item"
          >
            <span>
              Memo:
            </span>
            <p>
              795c52aa3017d3415c8da49b905e035e
            </p>
          </div>
          <button
            class="primary-button undefined"
            disabled=""
          >
            <div
              class="loading-indicator"
            >
              <svg
                fill="white"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12,4a8,8,0,0,1,7.89,6.7A1.53,1.53,0,0,0,21.38,12h0a1.5,1.5,0,0,0,1.48-1.75,11,11,0,0,0-21.72,0A1.5,1.5,0,0,0,2.62,12h0a1.53,1.53,0,0,0,1.49-1.3A8,8,0,0,1,12,4Z"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="0.75s"
                    repeatCount="indefinite"
                    type="rotate"
                    values="0 12 12;360 12 12"
                  />
                </path>
              </svg>
            </div>
            Transferring funds ...
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Bank modal test clicking overlay should close modal 1`] = `
<div>
  <div
    class="kima-modal bank-popup undefined "
  >
    <div
      class="modal-overlay"
    />
    <div
      class="modal-content-container"
    >
      <div
        class="kima-card-header"
      >
        <div
          class="topbar"
        >
          <div
            class="title"
          />
          <div
            class="control-buttons"
          >
            <button
              class="icon-button"
            >
              <svg
                height="32"
                viewBox="0 0 11 11"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.656 1.688L5.944 5.4L9.656 9.112L8.712 10.056L5 6.344L1.288 10.056L0.344 9.112L4.056 5.4L0.344 1.688L1.288 0.743999L5 4.456L8.712 0.743999L9.656 1.688Z"
                  fill="white"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="modal-content"
      >
        <div
          class="bank-simulation"
        >
          <h1>
            Welcome to the First National Crypto Bank
          </h1>
          <p>
            You’re about to send funds to the Kima Finance LTD. bank account
          </p>
          <div
            class="content-item"
          >
            <span>
              User account:
            </span>
            <p>
              2345234525245
            </p>
          </div>
          <div
            class="content-item"
          >
            <span>
              Sum:
            </span>
            <p>
              $
              0
            </p>
          </div>
          <div
            class="content-item"
          >
            <span>
              Memo:
            </span>
            <p>
              795c52aa3017d3415c8da49b905e035e
            </p>
          </div>
          <button
            class="primary-button undefined"
          >
            Submit
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
